torch/csrc/jit/passes/autocast.cpp

```diff
@@ -112,7 +106,7 @@ void castInputsToWidestType(Node* node) {
     if (auto tensor_type = input->type()->cast<TensorType>()) {
       const auto dtype = tensor_type->scalarType();
       if (!dtype.has_value() || *dtype != at::ScalarType::Half) {
-        castTensorInputs(node, at::ScalarType::Float);
+        castTensorInputs(node, aten::autocast_to_fp32);
```

Comment:
`*dtype != at::ScalarType::Half` <- isn't the type check here supposed to be Float instead?

Reply:
If we have any input dtype other than fp16 we'll attempt the autocast all inputs to fp32 (which will only actually cast fp16 tensors).

We could (and I'll probably do it in the next PR) check explicitly for `dtype == ScalarType::Float`, but the end result is equivalent.